### ISSUE TITLE
Condor submission failed at cmslpc

### ISSUE THREAD

**Issue by qliphy** [cms-sw/genproductions#2101]
@AndreasAlbert @kdlong 

It seems the gridpack condor submission script does not work at cmslpc for most of the nodes.
This might be due to "Moving to a central scheduling model"@cmslpc as mentioned [here](https://indico.cern.ch/event/744357/contributions/3075746/attachments/1775268/2889917/DMason_MBTonjes_LPCComputing_4January2019.pdf)

The error message is below:
      raise ClusterManagmentError, 'could not import htcondor python API: \n%s' % error
      madgraph.various.cluster.ClusterManagmentError: could not import htcondor python API: 

It seems we should update "source_condor.sh" to have HTCondor python bindings in submission step? 

Ref: https://hypernews.cern.ch/HyperNews/CMS/get/generators/4243/1/1/1/1/1/1.html

**Comment by qliphy** [comment]
@Saptaparna is also checking this.

**Comment by Saptaparna** [comment]
Yes, I have submitted a ticket, after verifying that condor submission still works for simple submission scripts. Will provide update based on feedback from LPC experts.

**Comment by Saptaparna** [comment]
The origin of the problem stems from the fact that condor_submit has changed under the hood. The condor_submit command is no longer the HTCondor command but the following:

exec env LD_LIBRARY_PATH='' /usr/bin/python -Ex "$0" "$@"
 
# The first line is necessary to work around LD_LIBRARY_PATH
# modifications done by CMSSW.
# -E ignores PYTHON envinronment variables set by USER
# -x ignores the first line of the script and treats the rest
# as python
#
# Keeping the python interpreter here as a reference
### !/usr/bin/python -E

This is the Fermilab specific wrapper around condor_submit for CMS LPC users.

**Comment by qliphy** [comment]
@Saptaparna  Thanks for the information!  Do you have a workaround?  And if possible can you update our script to make it work at cmslpc?

**Comment by Saptaparna** [comment]
@qliphy The obvious workaround is to update condor_submit with its cmslpc version. There is an added complication of making sure that schedd name is not hard coded and list of schedulers is provided (but this list may change over time). I am trying to deal with this now and following the suggestions of some of the experts here at the LPC.

**Comment by qliphy** [comment]
Suggestions from FNAL computing expert as below for your references:

---------------------

You CAN run on CMS Connect to T3_US_FNALLPC, so that's a really good point of Dave's, if everyone running gridpack just submits through CMS Connect, their jobs can run at T3_US_FNALLPC without rewriting the code.

One has to be sure that your CMS grid certificate is mapped to your FNAL username to be allowed in from CMS Connect and CRAB jobs, which I believe is true for all the gridpack users in this email.

-Marguerite
------------------------------------------------------------------------------------------
Dr. Marguerite Tonjes                                        
LPC Computing Support                       https://lpc.fnal.gov/computing
tonjes@fnal.gov                                          
Skype: phMarguerite            CMS Experiment Mattermost: @belt
office: (630) 840-2859                              FNAL WH11E        
------------------------------------------------------------------------------------------
